### PR TITLE
Swap burn send updates

### LINKED ISSUE #278: Funds might get stuck if on swap the routing contract does not use the full amount
If for any reason the routing contract does not use the full amount of the asset to swap for hats, the rest will stay in the contract, but not be accounted for in the balance still available to swap, this will make it stuck in the contract. Should we handle this case (considering that normally the swap contract is expected to use the full amount available) ?

### PR DESCRIPTION
**Pull request by ben-kaufman**
- Remove burn of swapped tokens
- Update amounts used correctly when router only swaps partial amount ( Fix #278 )

### CODE REVIEW

**Review by jellegerbrandy — APPROVED** [review]
looks good to me

File: contracts/HATVaultsRegistry.sol [hunk 1]
```diff
@@ -79,9 +77,8 @@ contract HATVaultsRegistry is Ownable {
     struct SwapData {
```
Comment (lirona) [inline contracts/HATVaultsRegistry.sol:77]: i know it's too late but i don't get why to create this struct if all it's members are just internal function variables and not passed anywhere.. makes more sense to me to just have them as local variables in the one function where you use it.. but it's not important so feel free to ignore :)

File: docs/roles.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ The `owner` of a hatvault created by the registry is the same registry owner.
 - can call `setVaultDescription` to change the vault's description hash
 - can call `setDepositPause` to pause and unpause deposits to the vault
 - can call `approveClaim` to approve a claim for a bounty payout that was previously submitted by a vault's committee
-- can call `swapBurnSend` and swap, burn, and send the vauit's token that are earmarked for payout (after approveClaim)
+- can call `swapAndSend` and swap and send the vauit's token that are earmarked for payout (after approveClaim)
```
Comment (lirona) [inline docs/roles.md:30]:
```suggestion
```

Comment (lirona) [reply docs/roles.md:30]: it's only called by the registry owner

**Review by lirona — COMMENTED** [review]
lgtm

File: contracts/HATVaultsRegistry.sol (hunk 1 above)
Comment (ben-kaufman) [reply contracts/HATVaultsRegistry.sol:77]: Because this allows to overcome the stack too deep error

Comment (jellegerbrandy) [reply contracts/HATVaultsRegistry.sol:94]: there's a comment in the code that is it to avoid stack too deep erros

Comment (lirona) [reply contracts/HATVaultsRegistry.sol:94]: ohhh ok missed that sorry

**Pull request merged**